cli/lib/kontena/client.rb

```diff
@@ -503,13 +503,12 @@ def add_version_warning(server_version)
 
     # Parse json
     #
-    # @param [String] json
+    # @param response [Excon::Response]
     # @return [Hash,Object,NilClass]
-    def parse_json(json)
-      JSON.parse(json)
+    def parse_json(response)
+      JSON.parse(response.body)
     rescue => ex
-      debug { "JSON parse exception: #{ex.class.name} : #{ex.message}" }
-      nil
+      raise Kontena::Errors::StandardError.new(400, "Invalid response JSON from server for #{response.path}: #{ex.class.name}: #{ex.message}")
```

Comment:
HTTP 400 = Bad Request

The request was perhaps fine, the response is bad. 

> 520 Unknown Error
> The 520 error is used as a "catch-all response for when the origin server returns something unexpected", listing connection resets, large headers, and empty or invalid responses as common triggers.

Reply:
Yeah... this is hacked to (ab)use `Kontena::Errors::StandardError` because that's the only way to get the CLI to print out anything semi-reasonable looking. There is no `exit_with_error` in the `Client` class..